build.gradle

```diff
@@ -1,53 +1,102 @@
 def protocVersion = protobufVersion
+
 dependencies {
     implementation "io.grpc:grpc-protobuf:${grpcVersion}"
-   implementation "io.grpc:grpc-services:${grpcVersion}"
-   implementation "io.grpc:grpc-stub:${grpcVersion}"
-   compileOnly "org.apache.tomcat:annotations-api:6.0.53"
-   runtimeOnly "io.grpc:grpc-netty-shaded:${grpcVersion}"
-   testImplementation "io.grpc:grpc-testing:${grpcVersion}"
-   testImplementation "io.grpc:grpc-inprocess:${grpcVersion}"
-   testImplementation('junit:junit:4.12')
-   testImplementation('org.junit.jupiter:junit-jupiter:5.10.1')
-	testImplementation('org.mockito:mockito-core:5.10.0')
+    implementation "io.grpc:grpc-services:${grpcVersion}"
+    implementation "io.grpc:grpc-stub:${grpcVersion}"
+    compileOnly "org.apache.tomcat:annotations-api:6.0.53"
+    runtimeOnly "io.grpc:grpc-netty-shaded:${grpcVersion}"
+    testImplementation "io.grpc:grpc-testing:${grpcVersion}"
+    testImplementation "io.grpc:grpc-inprocess:${grpcVersion}"
+    testImplementation('junit:junit:4.12')
+    testImplementation('org.junit.jupiter:junit-jupiter:5.10.1')
+    testImplementation('org.mockito:mockito-core:5.10.0')
 }
+
 protobuf {
-   protoc { artifact = "com.google.protobuf:protoc:${protocVersion}" }
-   plugins {
-       grpc { artifact = "io.grpc:protoc-gen-grpc-java:${grpcVersion}" }
-   }
-   generateProtoTasks {
-       all()*.plugins { grpc {} }
-   }
+    protoc { artifact = "com.google.protobuf:protoc:${protocVersion}" }
+    plugins {
+        grpc { artifact = "io.grpc:protoc-gen-grpc-java:${grpcVersion}" }
+    }
+    generateProtoTasks {
+        all()*.plugins { grpc {} }
+    }
 }
+
 test {
-   useJUnitPlatform()
+    useJUnitPlatform()
+}
+
+// Task to run the DataStoreServer
+task runDataStoreServer(type: JavaExec) {
+    group = 'application'
+    description = 'Starts the DataStoreServer.'
+    mainClass = 'com.example.clientandserver.DataStoreServer'
+    classpath = sourceSets.main.runtimeClasspath
+}
+
+// Task to run the FactorServer
+task runFactorServer(type: JavaExec) {
+    group = 'application'
+    description = 'Starts the FactorServer.'
+    mainClass = 'com.example.clientandserver.FactorServer'
+    classpath = sourceSets.main.runtimeClasspath
+}
+
+// Task to start both servers
+task runServersAndTest {
+   group = 'verification'
+    description = 'Starts both servers and runs tests.'
+
+    doLast {
+        // Start both servers asynchronously
+        def dataStoreProcess = ['java', '-cp', sourceSets.main.runtimeClasspath.asPath, 'com.example.clientandserver.DataStoreServer'].execute()
+        def factorServerProcess = ['java', '-cp', sourceSets.main.runtimeClasspath.asPath, 'com.example.clientandserver.FactorServer'].execute()
+
+        println "Both servers started for testing..."
+
+        try {
+            // Run the tests
+            exec {
+                commandLine './gradlew', 'test'
+            }
+        } finally {
+            // Stop both servers

```

Comment:
good job adding a command to stop the servers from continuing to run after the tests are done